CapstoneProject/Assets/Scripts/Core/Entity/Characters/Playable_Character/Player.cs

```diff
@@ -11,12 +11,14 @@ namespace Core.Entity
     [RequireComponent(typeof(PlayerStateComponent))]
     [RequireComponent(typeof(CollisionComponent))]
     [RequireComponent(typeof(PlayerAbilityComponent))]
+    [RequireComponent(typeof(CoinComponent))]
     public class Player : Character
     {
         public PlayerStateComponent States;
         public CollisionComponent CollisionComponent;
         public PlayerAbilityComponent AbilityComponent;
         public PlayerHealthComponent HealthComponent;
+        public CoinComponent CoinComponent;
```

Comment:
if players and enemies both have coin components, would it not be better to just put this in the character or is there another type of character?